tests/test_singularity.py

```diff
@@ -13,6 +14,42 @@
 sys.argv = ['']
 
 
+@needs_singularity
+def test_singularity_pullfolder(tmp_path):
+    for directory, subdir, files in os.walk(os.getcwd()):
+        for image in files:
+            if image.startswith("debian"):
+                os.remove(os.path.join(directory, image))
+                print("removed image hiding away deep within!")
+
+    if "SINGULARITY_PULLFOLDER" in os.environ:
+        del os.environ["SINGULARITY_PULLFOLDER"]
```

Comment:
Great, now that the tests work we can refactor `get_main_output` to include a custom `env` instead of deleting from the global `os.environ`

Reply:
I can help with that

Reply:
This sounds good!